frontend/src/main/components/Commons/CommonsForm.js

```diff
@@ -249,14 +256,39 @@ function CommonsForm({initialCommons, submitAction, buttonLabel = "Create"}) {
                     isInvalid={!!errors.startingDate}
                     {...register("startingDate", {
                         valueAsDate: true,
-                        validate: {isPresent: (v) => !isNaN(v)},
+                        validate: {
+                            isPresent: (v) => !isNaN(v) || "Start date is required" , 
+                            // Stryker disable next-line all
+                            crossVal: () => Number(getValues("startingDate")) <= Number(getValues("lastDay")) || "Start date must be on or before last day"}, 
+                
                     })}
                 />
                 <Form.Control.Feedback type="invalid">
                     {errors.startingDate?.message}
                 </Form.Control.Feedback>
             </Form.Group>
             
+            <Form.Group className="mb-5" style={{width: '300px', height: '50px'}} data-testid={`${testid}-r4`}>
+                <Form.Label htmlFor="lastDay">Last Day</Form.Label>
+                <Form.Control
+                    data-testid={`${testid}-lastDay`}
+                    id="lastDay"
+                    type="date"
+                    defaultValue={DefaultVals.lastDay}
+                    isInvalid={!!errors.lastDay}
+                    {...register("lastDay", { 
+                        valueAsDate: true,
+                        validate: {
+                            isPresent: (v) => !isNaN(v) || "Last Day is required"  
+                            //crossVal: () => getValues(getValues("startingDate")) <= getValues(getValues("lastDay")) || "Last Day must come after starting date"
+                    
+                        },
+                    })}
+                />
+                <Form.Control.Feedback type="invalid">
+                    {errors.lastDay?.message}
+                </Form.Control.Feedback>
+            </Form.Group>
```

Comment:
You may want to consider using the `<Row><Col>` construct that you find earlier in the file to get the Start Date and End Date side by side instead of on top of one another.  That would make the form look a bit more compact.

Reply:
Good Idea! the change has been implemented